contracts/oracles/WstETHOracle.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.13;
+
+import "../interfaces/OracleInterface.sol";
+import "../interfaces/IStETH.sol";
+
+/**
+ * @title WstETHOracle
+ * @author Venus
+ * @notice This oracle fetches the price of wstETH asset
+ */
+contract WstETHOracle is OracleInterface {
+    /// @notice Price denominator of WETH/USD price returned from Resilient Oracle
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public constant WETH_USD_PRICE_DENOMINATOR = 1e18;
+
+    /// @notice Address of stETH
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IStETH public immutable STETH;
+
+    /// @notice Address of wstETH
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    address public immutable WSTETH_ADDRESS;
+
+    /// @notice Address of WETH
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    address public immutable WETH_ADDRESS;
+
+    /// @notice Address of Resilient Oracle
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    OracleInterface public immutable RESILIENT_ORACLE;
+
+    modifier notNullAddress(address someone) {
```

Comment:
We could use instead the function `ensureNonzeroAddress`, from https://github.com/VenusProtocol/solidity-utilities/blob/develop/contracts/validators.sol#L13